### PR TITLE
ScalaSerialDescriptor -> SerialDescriptor

### PR DESCRIPTION
**Pull request by Yawolf**
- Now `ScalaSerialDescriptor` is called just `SerialDescriptor`
- `ScalaSerialDescriptorContext` has been renamed to `SerialDescriptorInstances`, and now is a class from which extends the `SerialDescriptor` companion object in order to avoid given imports.
- Docs updated
- Added two `TODO` regarding the undefined implementation of `serialize` and `deserialize` methods in `KSerializer` instance.

### CODE REVIEW

File: scala/src/main/scala/com/xebia/functional/xef/scala/auto/SerialDescriptorContext.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package com.xebia.functional.xef.scala.auto
```
Comment (necosta) [inline scala/src/main/scala/com/xebia/functional/xef/scala/auto/SerialDescriptorContext.scala:1]: Rename file?

**Review by necosta — COMMENTED** [review]
I think you have to rebase as merging does not fix the issues on file SynopsisReview and PromptTemplate already merged into main.

**Comment by necosta** [comment]
The changes look good, happy to approve after rename conflicts get solved.

File: scala/src/main/scala/com/xebia/functional/xef/scala/auto/SerialDescriptorContext.scala (hunk 1 above)
Comment (Yawolf) [reply scala/src/main/scala/com/xebia/functional/xef/scala/auto/SerialDescriptorContext.scala:1]: Oh, right!

Comment (Yawolf) [reply scala/src/main/scala/com/xebia/functional/xef/scala/auto/SerialDescriptorContext.scala]: Done!

**Review by necosta — APPROVED** [review]

**Review by juanpedromoreno — APPROVED** [review]

**Review by juanpedromoreno — APPROVED** [review]

**Pull request merged**